bluesky/plan_patterns.py

```diff
@@ -62,6 +62,120 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
     cyc += cycler(y_motor, y_points)
     return cyc
 
+        
+def spiral_square_pattern(x_motor, y_motor, x_centre, y_centre, x_range, y_range, x_num,y_num):
+    '''Square spiral scan, centered around (x_start, y_start)
+    Parameters
+    ----------
+    x_motor : object, optional
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object, optional
+        any 'setable' object (motor, temp controller, etc.)
+    x_centre : float
+        x center
+    y_centre : float
+        y center
+    x_range : float
+        x width of spiral
+    y_range : float
+        y width of spiral
+    x_num : float
+        number of x axis points
+    y_num : float (must be even if x_num is even and must be odd if x_num is odd, if not it is increased by 1 to ensure this)
+        number of y axis points 
+    Returns
+    -------
+    cyc : cycler
+    '''
+
+
+    x_points, y_points = [], []
+
+    if x_num%2==0:
+        num_st= 2
+        if y_num%2==1:
+           y_num+=1
+
+        offset=0.5
+        
+    else:
+        num_st=1
+        if y_num%2==0:
+            y_num+=1
+
+        offset=0
+        
+        x_points.append(x_centre)
+        y_points.append(y_centre)
+        
+    #print( 'x_range = '+str(x_range)+', x_num = '+str(x_num)  )
```

Comment:
It's a minor thing, but please delete the comment print statements and the blocks of blank lines. (For readability, use occasional blank lines to taste but avoid blocks like the three blank lines below.)